pkg/cri/opts/spec_windows.go

```diff
@@ -142,6 +142,16 @@ func WithWindowsMounts(osi osinterface.OS, config *runtime.ContainerConfig, extr
 				if dst[0] == '\\' {
 					dst = "C:" + dst
 				}
+				// filepath.Clean adds a '.' at the end if the path is a
+				// drive (like Z:, E: etc.). Keeping this '.' in the path
+				// causes incorrect parameter error when starting the
+				// container on windows. Same happens if the drive path ends with a '\'.
+				// Remove it here.
+				// No other valid path after filepath.Clean should have only
+				// 3 letters with ':' as the 2nd char.
+				if len(dst) == 3 && dst[1] == ':' {
```

Comment:
I think it would be clearer to check up above if the path is a drive letter (e.g. `D:`), and skip the `Clean` call if so.

Reply:
I specifically avoided that because someone can still pass a path like `D:\.\.\` which would pass our check of not being a drive letter, go through filepath.Clean and will still cause failures.

Reply:
`D:\.\.\` should clean to `D:\` which I expect would work fine as a mount path. Does that not work?

Reply:
Yeah, even `D:\` would fail with the same error.